specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json

```diff
@@ -1377,6 +1377,109 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/staticSites/{name}/getBasicAuthState": {
+      "post": {
+        "tags": [
+          "StaticSites"
+        ],
+        "summary": "Gets the basic auth state of a static site.",
+        "description": "Description for Gets the basic auth state of a static site.",
+        "operationId": "StaticSites_GetBasicAuth",
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "description": "Name of the static site.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/StaticSiteBasicAuthPropertiesARMResource"
+            }
+          },
+          "default": {
+            "description": "App Service error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get the basic auth state of a static site": {
+            "$ref": "./examples/GetStaticSiteBasicAuthState.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/staticSites/{name}/createOrUpdateBasicAuth": {
```

Comment:
>createOrUpdateBasicAuth [](http://example.com/codeflow?start=115&length=23)

do u not intend to have a delete API?

Reply:
I don't, basic auth isn't a resource as much as it is a state of a static web app, so a create or update call with an empty value serves the function of a delete call.

Reply:
@rkmanda Let me know if this isn't acceptable, but I'll resolve the comment for now as this was intentional.